src/main/java/edu/indiana/dlib/amppd/service/impl/AmpUserServiceImpl.java

```diff
@@ -439,6 +439,13 @@ public AmpUser getUser(String username) {
 		
 		return null;
 	}
+	
+	public AmpUser getUserById(Long userId) {
+		Optional <AmpUser> userOpt= ampUserRepository.findById(userId);
+		if(userOpt.isPresent()) return userOpt.get();
+		return null;
```

Comment:
When ID is invalid, we should log an error msg. In general for most service methods,  we put log info near the end for a successful operation, and error msg if exception or error. For complicated operations, we should put log at each important phase within the method. Please observe other service methods for this pattern.